File: typescript-examples/src/module1/testDoubles/dummy/OrderProcessor.ts
import {User} from "./User.ts";

export class OrderProcessor {
    private orderCount: number;

    constructor(orderCount: number) {
        this.orderCount = orderCount;
    }

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    public processOrder(_user: User): number {
        this.orderCount++;

        const orderId = this.orderCount;

        console.log("Processing order " + orderId);

        return orderId;
    }
}

File: src/main/java/com/codurance/module1/TestDoubles/Dummy/OrderProcessor.java
package com.codurance.module1.TestDoubles.Dummy;

public class OrderProcessor {

    private int orderCount;

    OrderProcessor(int orderCount) {

        this.orderCount = orderCount;
    }

    public int processOrder(User user) {
        orderCount++;

        int orderId = orderCount;

        // The user object is not used in this test
        System.out.println("Processing order " + orderId);

        return orderId;
    }

}
